### PR TITLE
Northwest Class 4 - Humail Khan - JavaScript-Core-1 - Coursework - Week 1

### PR DESCRIPTION
**Pull request by humailhasankhan**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: exercises/G-numbers/exercise.js [hunk 1]
```diff
@@ -1 +1,8 @@
 // Start by creating a variables `numberOfStudents` and `numberOfMentors`
+const numberOfStudents = 15;
+const numberOfMentors = 8;
+const totalNumbers = numberOfStudents + numberOfMentors;
+
+console.log(`Number of students: ${numberOfStudents}`);
```
Comment (peedeerich) [inline exercises/G-numbers/exercise.js:6]: nice use of string template

File: extra/2-piping.js [hunk 2]
```diff
@@ -16,26 +16,27 @@
...
-
+function format(amount) {
+  return `£${parseFloat(amount.toFixed(1))}`;
 }
 
 const startingValue = 2;
 
 // Why can this code be seen as bad practice? Comment your answer.
-let badCode =
+let badCode = format(multiply(add(startingValue, 10), 2));
```
Comment (peedeerich) [inline extra/2-piping.js:34]: Can you articulate why this code would be bad practice?

File: extra/3-magic-8-ball.js [hunk 3]
```diff
@@ -58,9 +99,21 @@ function shakeBall() {
...
   //Write your code in here
+
+  if (veryPositive.includes(answer)) {
+    return "very positive";
+  } else if (positive.includes(answer)) {
+    return "positive";
+  } else if (negative.includes(answer)) {
+    return "negative";
+  } else if (veryNegative.includes(answer)) {
+    return "very negative";
```
Comment (peedeerich) [inline extra/3-magic-8-ball.js:113]: nice - this is a good use case for the array method "includes"

File: mandatory/3-function-output.js [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
+//This function takes two parameters which are strings and concatenates them into one string.
 function combine2Words(word1, word2) {
   return word1.concat(word2);
 }
 
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  // We could not use concat() because it only accepts strings. It would not meet all conditions of the test.
+  return firstWord + " " + secondWord + " " + thirdWord;
```
Comment (peedeerich) [inline mandatory/3-function-output.js:17]: How would you explain what happens in this function in order for the last test case to work with a number? 

Google "javascript coercion" if you'd like to find out more! We'll see more examples of this throughout the course.

**Comment by peedeerich** [comment]
Excellent work - well done! You have also used some array methods I see - you'll be seeing more of those in JS 3 so stay tuned :)

File: exercises/G-numbers/exercise.js (hunk 1 above)
Comment (humailhasankhan) [reply exercises/G-numbers/exercise.js:6]: Thank you.

File: extra/2-piping.js (hunk 2 above)
Comment (humailhasankhan) [reply extra/2-piping.js:34]: My take on this one was that chaining too many times would be considered as bad practice.

File: extra/3-magic-8-ball.js (hunk 3 above)
Comment (humailhasankhan) [reply extra/3-magic-8-ball.js:113]: Thank you Paul. This was refactored with the help of Guardians.

File: mandatory/3-function-output.js (hunk 4 above)
Comment (humailhasankhan) [reply mandatory/3-function-output.js:17]: I would have to look it up. But my guess is that it has to do something with adding numbers and strings together. 13 is a number in the last test and the rest are strings. Probably the typeof would be different and they would be added together but not computed? I will look into javascript coercion. Thanks Paul.

**Pull request closed without merging**